pkg/verify/verify.go

```diff
@@ -0,0 +1,198 @@
+//
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verify
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/cyberphone/json-canonicalization/go/src/webpki.org/jsoncanonicalizer"
+	"github.com/sigstore/rekor/pkg/generated/client"
+	"github.com/sigstore/rekor/pkg/generated/client/tlog"
+	"github.com/sigstore/rekor/pkg/generated/models"
+	"github.com/sigstore/rekor/pkg/util"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/options"
+	"github.com/transparency-dev/merkle/proof"
+	"github.com/transparency-dev/merkle/rfc6962"
+)
+
+// ProveConsistency verifies consistency between an initial, trusted STH
+// and a second new STH. Assumes that the signature on the STHs' are verified.
```

Comment:
I wonder if it'd be better to check the signatures on the STHs, this seems like something that'd be easy to skip as a client. Same with `VerifyCurrentCheckpoint` - It verifies the newest checkpoint, but not the previous.

Reply:
The previous/old one should already be trusted in `VerifyCurrentCheckpoint`, but I can add for rigor

Reply:
Note that for that, we would need to pass in the verifier. I think including this in the doc string is fine?

Reply:
Yea, I'm ok with that. An example somewhere of a few different ways of composing these functions might be useful

Reply:
Discussed, examples to be added elsewhere.